gensim/models/doc2vec.py

```diff
@@ -42,48 +42,67 @@
 except ImportError:
     from Queue import Queue
 
-from numpy import zeros, random, sum as np_sum
-from six import string_types
+from collections import namedtuple
+
+from numpy import zeros, random, sum as np_sum, add as np_add, concatenate, \
+    repeat as np_repeat, array, float32 as REAL, empty, ones, memmap as np_memmap, \
+    sqrt, newaxis, ndarray, dot, argsort, vstack
 
 logger = logging.getLogger(__name__)
 
-from gensim import utils  # utility fnc for pickling, common scipy operations etc
-from gensim.models.word2vec import Word2Vec, Vocab, train_cbow_pair, train_sg_pair
+from gensim import utils, matutils  # utility fnc for pickling, common scipy operations etc
+from gensim.models.word2vec import Word2Vec, Vocab, train_cbow_pair, train_sg_pair, train_sentence_sg
+from six.moves import xrange
+from six import string_types, integer_types
 
 try:
-    from gensim.models.doc2vec_inner import train_sentence_dbow, train_sentence_dm, FAST_VERSION
+    from gensim.models.doc2vec_inner import train_sentence_dbow, train_sentence_dm, train_sentence_dm_concat,\
+                                            FAST_VERSION
 except:
     # failed... fall back to plain numpy (20-80x slower training than the above)
     FAST_VERSION = -1
 
-    def train_sentence_dbow(model, sentence, lbls, alpha, work=None, train_words=True, train_lbls=True):
+    def train_sentence_dbow(model, word_vocabs, doctag_indices, alpha, work=None,
+                            train_words=False, learn_doctags=True, learn_words=True, learn_hidden=True,
+                            word_vectors=None, word_locks=None, doctag_vectors=None, doctag_locks=None):
         """
         Update distributed bag of words model by training on a single sentence.
 
         The sentence is a list of Vocab objects (or None, where the corresponding
```

Comment:
I see. Yes, that makes sense.

And yes, consistency in naming will make the code / API easier to follow :+1: